Products/CMFPlone/patterns/__init__.py

```diff
@@ -14,6 +14,7 @@
 from plone.app.theming.utils import theming_policy
 from Products.CMFCore.utils import getToolByName
 from zope.component.hooks import getSite
+from plone import api
```

Comment:
We are not supposed to use `plone.api` in core packages (to avoid circular dependencies).
Moreover, it looks like the portal url is already available in `generator.portal_url`.